src/main/java/org/idea/plugin/atg/index/XmlActorIndexExtension.java

```diff
@@ -0,0 +1,96 @@
+package org.idea.plugin.atg.index;
+
+import com.intellij.ide.highlighter.XmlFileType;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiManager;
+import com.intellij.util.indexing.*;
+import com.intellij.util.io.DataExternalizer;
+import com.intellij.util.io.EnumeratorStringDescriptor;
+import com.intellij.util.io.KeyDescriptor;
+import org.idea.plugin.atg.visitor.XmlPsiRecursiveElementVisitor;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class XmlActorIndexExtension extends FileBasedIndexExtension<String, String> {
```

Comment:
What purpose of given index? 
Every additional index slows down when any file changes(just one symbol typed), as all indexes are considered for being rebuilt.
At this moment I see, you just use it as list of ATG-related Xml files (by standard it should be ATG config folders) which have specific tag. As you anyway invokes parsing of all these files I'd recommend to use replace this index with `AtgXmlsIndexExtension` (which already indexes all XMLs located under Config/ConfigLayer folders) + additional filter of top XML tag. There shouldn't be huge amount of XML files
PS. I don't understand purpose of storing fileName/path there.